R2API/Utils/Reflection.cs

```diff
@@ -923,5 +925,81 @@ public static class Reflection {
         }
 
         #endregion
+
+        public static System.Reflection.FieldInfo GetNestedField(Type type, string fieldName) {
+            var nestedTypes = type.GetNestedTypes((System.Reflection.BindingFlags)(-1));
+            foreach (Type nestedType in nestedTypes) {
+                var fieldInfo = nestedType.GetField(fieldName, System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+                if (fieldInfo != null) {
+                    return fieldInfo;
+                }
+                fieldInfo = nestedType.GetField(fieldName, System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static);
+                if (fieldInfo != null) {
+                    return fieldInfo;
+                }
+                fieldInfo = nestedType.GetField(fieldName, System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Instance);
+                if (fieldInfo != null) {
+                    return fieldInfo;
+                }
+                fieldInfo = nestedType.GetField(fieldName, System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static);
+                if (fieldInfo != null) {
+                    return fieldInfo;
+                }
+            }
+            return null;
+        }
+
+        public static System.Reflection.MethodInfo GetNestedMethod(Type type, string methodName) {
+            var nestedTypes = type.GetNestedTypes((System.Reflection.BindingFlags)(-1));
+            foreach (Type nestedType in nestedTypes) {
+                var methodInfo = nestedType.GetMethod(methodName, System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+                if (methodInfo != null) {
+                    return methodInfo;
+                }
+                methodInfo = nestedType.GetMethod(methodName, System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Static);
+                if (methodInfo != null) {
+                    return methodInfo;
+                }
+                methodInfo = nestedType.GetMethod(methodName, System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Instance);
+                if (methodInfo != null) {
+                    return methodInfo;
+                }
+                methodInfo = nestedType.GetMethod(methodName, System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static);
+                if (methodInfo != null) {
+                    return methodInfo;
+                }
+            }
+            return null;
+        }
+        public static void LogCursorOpcodes(ILCursor cursor) {
```

Comment:
This modifies the cursor passed

Reply:
True. I should probably restore the cursor index to its prior position when the function finishes.